### ISSUE TITLE
Skip fiber-geometry model

### ISSUE THREAD

**Issue by oesteban** [ecaruyer/phantomas#29]
Hi @ecaruyer, is it possible with/without modifications of phantomas to simulate the dwi signal from dyads as generated by Bedpost? This means: we have voxel-wise information of the orientations of up-to 3 fibers, and then the corresponding volume of each.

**Comment by ecaruyer** [comment]
Hi Oscar @oesteban , Sorry for the late reply. I don't know the exact output of Bedpost, but I would say that with a bit of tweaking it should be possible. You'll have to fix the kind of forward model and diffusion parameters once for every compartments. Things are defined in https://github.com/ecaruyer/phantomas/blob/master/phantomas/mr_simul/synthetic.py, I created an abstract class, called AxiallySymmetricModel, defining an interface to generate a diffusion signal. So far, there is only GaussianModel implementing this abstract class, but I must have some code under the hood for restricted diffusion. Let me know if this fits your needs. Cheers!

**Comment by oesteban** [comment]
Hi there @ecaruyer, I'm coming back to this :).

So, I have 3 maps of 3D sticks (vectors, or dyads using FSL's nomenclature) with their corresponding volume fraction. I also want to be able to simulate up to 3 isotropic compartments with different diffusivities.

Then, I could extend the ```AxiallySymmetricModel``` for my model (say *balls and tensors* model), right?. Would it be enough to implement the ```signal``` member only or there is something else?.

Basically, I would need to implement [this](https://github.com/ecaruyer/phantomas/blob/master/phantomas/mr_simul/synthetic.py#L134) for each of the three sticks, and then the signal loss due to the isotropic fractions, isn't it?

**Comment by oesteban** [comment]
Mmm, I see that I will also need to modify the convolution right?

**Comment by oesteban** [comment]
OR, just replace L196-200 (https://github.com/ecaruyer/phantomas/blob/master/scripts/phantomas_dwis#L196):

Here, I could supply my 3D sticks in ```fod_samples``` with their corresponding volume fraction in ```fod_weights``` at the sub-grid resolution, right?

Are ```fod_samples``` equivalent to my *sticks* or should I generate it from the sticks (peaks of a prolate tensor)?

**Comment by ecaruyer** [comment]
Hi Oscar @oesteban, 

Yes, phantomas uses a continuous representation of the FOD, so first you would have to convert your sticks (of up to 3 fibers) to an fod using phantomas.mr_simul.fod.compute_fod. Then you would have to synthesize the diffusion signal following the same logic as there https://github.com/ecaruyer/phantomas/blob/master/scripts/phantomas_dwis#L217-L236.

Hope this helps,
Manu.

**Comment by oesteban** [comment]
I have opened #30 to implement this feature, let's keep discussion there :). Thanks a lot @ecaruyer .
